### PR TITLE
Use media query

### PR DESCRIPTION
**Pull request by orrgottlieb**
#### Basic
- [ ] Used plop (`npm run plop`) to create a new component.
- [ ] PR has description.
- [ ] New component is functional and uses Hooks. 
- [ ] Component defines [`PropTypes`](https://reactjs.org/docs/typechecking-with-proptypes.html).
#### Style
- [ ] Styles are added to `NewComponent.modules.scss` file inside of the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/).
#### Storybook
- [ ] Stories were added to `/src/NewComponent/__stories__/NewComponent.stories.js` file.
- [ ] Stories include all flows of using the component.
#### Tests
- [ ] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

### CODE REVIEW

File: src/hooks/useMediaQuery/__stories__/useMediaQuery.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+- [Feedback](#feedback)
+
+## Overview
+
+This hook helps maps if a media query is matched or not
+
+<Canvas>
+  <Story name="Single Rule">
+    {() => {
+      const [mediaQueryIsMatching] = useMediaQuery(["screen and (max-width: 1023px) and (min-width: 768px)"]);
```
Comment (hadasfa) [inline src/hooks/useMediaQuery/__stories__/useMediaQuery.stories.mdx:21]: cool! not sure if string API is better then more structure API( attribute for media type, attribute for media features)

File: src/hooks/useMediaQuery/__stories__/useMediaQuery.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      return (
+        <div>
+          media query - "screen and (max-width: 1023px) and (min-width: 768px)" is matching:{" "}
```
Comment (hadasfa) [inline src/hooks/useMediaQuery/__stories__/useMediaQuery.stories.mdx:24]: I guess {""} is a typo right?

File: src/hooks/useMediaQuery/__stories__/useMediaQuery.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        "prefers-color-scheme: dark"
+      ]);
+      return (
+        <div>
+          <div>
+            media query - "screen and (max-width: 1280px) and (min-width: 768px)" is matching:{" "}
+            {screenSizeMediaQuery ? "true" : "false"}
+          </div>
+          <div>
+            media query - prefers-color-scheme: dark is matching:{" "}
```
Comment (hadasfa) [inline src/hooks/useMediaQuery/__stories__/useMediaQuery.stories.mdx:46]: {""} is a typo?

File: src/hooks/useMediaQuery/__stories__/useMediaQuery.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+          </div>
+        </div>
+      );
+    }}
+  </Story>
+</Canvas>
+
+## Arguments
+
+<FunctionArguments>
```
Comment (hadasfa) [inline src/hooks/useMediaQuery/__stories__/useMediaQuery.stories.mdx:57]: I think it's critical to add here also a documentation of the return value of the hooks as well, like in other hooks (there is few examples for it in our storybook)

File: src/hooks/useVibeMediaQuery/index.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { useLayoutEffect, useState } from "react";
+import useMediaQuery from "hooks/useMediaQuery";
+import { MEDIA_QUERY_SIZES, LARGE, SMALL1, XLARGE, MEDIUM1, MEDIUM2, SMALL2 } from "utils/media-query-utils";
+
+export default function useVibeMediaQuery() {
+  const [mediaSize, setMediaSize] = useState(SMALL1);
+  const [isSmall1, isSmall2, isMedium1, isMedium2, isLarge, isXLarge] = useMediaQuery(MEDIA_QUERY_SIZES);
+
+  useLayoutEffect(() => {
+    if (isSmall1) setMediaSize(SMALL1);
```
Comment (hadasfa) [inline src/hooks/useVibeMediaQuery/index.js:10]: why not use terminology which related to the kind of screen we are using? Ipad ? Phone? Computer screen?

File: src/hooks/useVibeMediaQuery/__stories__/useVibeMediaQuery.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+- [Arguments](#arguments)
+- [Feedback](#feedback)
+
+## Overview
+
+This hook will return the value of the current vibe break point
+
+| Value | Media query |
+|-------|-------------|
+| 1     | screen and (max-width: 767px)            |
```
Comment (hadasfa) [inline src/hooks/useVibeMediaQuery/__stories__/useVibeMediaQuery.stories.mdx:21]: Why not writing here the names of the enums the same as in the code?

File: src/hooks/useMediaQuery/index.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  const queries = useMemo(() => {
+    return Array.isArray(query) ? query : [query];
+  }, [query]);
+
+  const [matches, setMatches] = useState(queries.map(query => !!window.matchMedia(query).matches));
+
+  useLayoutEffect(() => {
+    const mediaQueryList = queries.map(query => window.matchMedia(query));
+    const updated = mediaQueryList.map((query, index) => {
+      const callback = event => {
```
Comment (hadasfa) [inline src/hooks/useMediaQuery/index.js:13]: maybe add a small comment about this callback ? that we would like to listen to changes on the window and update this state id this query match / not match after each update?

File: src/hooks/useMediaQuery/__stories__/useMediaQuery.stories.mdx (hunk 2 above)
Comment (orrgottlieb) [reply src/hooks/useMediaQuery/__stories__/useMediaQuery.stories.mdx:24]: no, it was done automatically by the ide - it is spacing

File: src/hooks/useVibeMediaQuery/index.js (hunk 5 above)
Comment (orrgottlieb) [reply src/hooks/useVibeMediaQuery/index.js:10]: we are not targeting specific types - as they change and there is no clear definition for what it is a tables/phone...., more than that - you can change orientation in tables, phones and resize screen so it is not relevant

File: src/hooks/useVibeMediaQuery/__stories__/useVibeMediaQuery.stories.mdx (hunk 6 above)
Comment (orrgottlieb) [reply src/hooks/useVibeMediaQuery/__stories__/useVibeMediaQuery.stories.mdx:21]: this is the value which returns from the function

**Review by hadasfa — APPROVED** [review]

File: src/hooks/useVibeMediaQuery/__stories__/useVibeMediaQuery.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+- [Returns](#returns)
+- [Feedback](#feedback)
+
+## Overview
+
+This hook will return the value of the current vibe break point
+
+| Value | Media query                                            |
+| ----- | ------------------------------------------------------ |
+| 1     | screen and (max-width: 767px)                          |
```
Comment (hadasfa) [inline src/hooks/useVibeMediaQuery/__stories__/useVibeMediaQuery.stories.mdx:21]: maybe just add here the enum key also ?

File: src/hooks/useMediaQuery/__stories__/useMediaQuery.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+## Returns
+
+<FunctionArguments>
+  <FunctionArgument
+    name="result"
+    type="Array<boolean>"
+    description={
+      <>
+        An array with the matching media query set to true for example:
+        <br /> <code>[true, false]</code> - the first media query applies and the second is not
```
Comment (hadasfa) [inline src/hooks/useMediaQuery/__stories__/useMediaQuery.stories.mdx:72]: great!

**Pull request merged**